File: Java/src/Engine/GameField.java
/**
 * author Z3R0ing
 */
package Engine;

import java.awt.Color;
import java.awt.Graphics;
import java.awt.Image;
import java.util.ArrayList;

import javax.swing.JPanel;

import Data.Files;
import SaveSamurai.Main;

/**
 * Панель игрового поля, на которой рисуются все объекты
 */
public class GameField extends JPanel {

	public static final int GAME_FIELD_WIDTH = Main.FRAME_WIDTH - 6;
	public static final int GAME_FIELD_HEIGHT = Main.FRAME_HEIGHT - 29;

	///
	private Image background;

	/// Список элементов на игровом поле
	private ArrayList<Drawable> elements;

	/**
	 * 
	 */
	public GameField() {
		// Задаём цвет фона
		setBackground(Color.WHITE);
		// Инициализируем список элементов
		elements = new ArrayList<Drawable>();
		background = Files.getTexture("back");
	}

	/**
	 * Добавляет элемент на поле
	 * 
	 * @param element
	 *            - элемент
	 */
	public void add(Drawable element) {
		elements.add(element);
	}

	/**
	 * Отрисовывает игровое поле
	 */
	public void paintComponent(Graphics gr) {
		// Очищаем панель
		gr.clearRect(0, 0, 800, 600);
		// Рисуем фон
		gr.drawImage(background, 0, 0, null);
		// Отрисовываем игровые элементы
		for (Drawable element : elements) {
			if (element != null)
				element.draw(gr);
		}
	}

}

File: C Sharp/Engine/GameField.cs
using SaveSamurai.Data;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

namespace SaveSamurai.Engine
{
    public class GameField : Panel
    {
        public static readonly int GAME_FIELD_WIDTH = Program.FRAME_WIDTH - 6;
        public static readonly int GAME_FIELD_HEIGHT = Program.FRAME_HEIGHT - 29;

        ///
        private Image background;

        /// Список элементов на игровом поле
        private List<Drawable> elements;

        /// Переменная графики
        private Graphics gr;

        public GameField()
        {
            DoubleBuffered = true;
            Dock = DockStyle.Fill;
            Location = new Point(0, 0);
            Name = "GameField";
            Size = new Size(784, 561);
            TabIndex = 0;
            Paint += new PaintEventHandler(GameField_Paint);
            gr = CreateGraphics();
            elements = new List<Drawable>();
            background = Files.getTexture("back");
        }

        /**
	 * Добавляет элемент на поле
	 * 
	 * @param element
	 *            - элемент
	 */
        public void add(Drawable element)
        {
            elements.Add(element);
        }

        /**
         * Отрисовывает игровое поле
         */
        private void GameField_Paint(object sender, PaintEventArgs e)
        {
            gr = e.Graphics;
            // Очищаем панель
            gr.Clear(BackColor);
            // Рисуем фон
            gr.DrawImage(background, 0, 0);
            // Отрисовываем игровые элементы
            foreach (Drawable element in elements)
            {
                if (element != null)
                    element.draw(gr);
            }
        }
    }
}
